src/java.base/share/classes/java/time/OffsetDateTime.java

```diff
@@ -167,17 +167,26 @@ public static Comparator<OffsetDateTime> timeLineOrder() {
      * Compares this {@code OffsetDateTime} to another date-time.
      * The comparison is based on the instant.
      *
+     * When two values represent the same instant, the local date-time is compared
+     * to distinguish them. This step is needed to make the ordering
+     * consistent with {@code equals()}.
+     *
```

Comment:
This method is also called from the public `timeLineOrder()` method. Could this change cause a different order?

Reply:
Your are right; I missed that subtlety. With that in mind I don't think the performance impact is sufficient to duplicate more code.